File: connected-devices-python workspace/iot-device/apps/labs/module06/MqttSubClientTestApp.py
'''
Created on Mar 2, 2019

@author: khars
'''

'''
This App consists of instance of MqttClientConnector and Subscribes to a topic and
retrieves published message so its a MQTT Subscriber Client
'''
from labs.common import ConfigConst
from labs.common.ConfigUtil import ConfigUtil
from labs.module06.MqttClientConnector import MqttClientConnector
from labs.common.DataUtil import DataUtil
import logging

'''
Setting values for Topic and address for MQTT broker
'''
topic = "Temperature Sensor"
config = ConfigUtil('../../../config/ConnectedDevicesConfig.props');
host = config.getProperty(ConfigConst.ConfigConst.MQTT_CLOUD_SECTION, ConfigConst.ConfigConst.HOST_KEY)
subscribe = MqttClientConnector(topic)
subscribe.subscribe(host)                  # Connecting to MQTT Broker
msg = subscribe.message()                  # Subscribing to specefied Topic
logging.debug('JSon Data Received:')
print("Json Data Received:\n "+str(msg)+"\n")
data = DataUtil();
sensor = data.jsonToSensorData(msg)        # Converting Json data to Sensor Data
logging.debug('Json data converted into SensorData')
print("Received message in SensorData format"+str(sensor))
json = data.SensorDataToJson(sensor)
logging.debug('SensorData converted into Json Data: ')
print('SensorData converted to Json Data again: \n'+str(json))

File: connected-devices-java workspace/iot-gateway/src/main/java/schooldomain/studentname/connecteddevices/labs/module06/MqttSubClientTestApp.java
package schooldomain.studentname.connecteddevices.labs.module06;

/*
 * This Class is used to Subscribe to topic from MQTT Broker 
 * 
 * @author - Venkat Prasad Krishnamurthy
 */

import java.util.logging.Level; 
import java.util.logging.Logger;
import com.labbenchstudios.edu.connecteddevices.common.ConfigConst;
import schooldomain.studentname.connecteddevices.common.DataUtil;
import schooldomain.studentname.connecteddevices.common.SensorData;

public class MqttSubClientTestApp {
	
	private static final Logger logger = Logger.getLogger(MqttSubClientTestApp.class.getName());
	private static MqttSubClientTestApp app;
	private MqttClientConnector mqttClient;
	public MqttSubClientTestApp()
	 {
	 super();
	 }
	
	/*
	 * method used to initialize subscribe action
	 * 
	 * @param topicName: Topic to be subscribed
	 */
	public void start(String topicName)
	{
		mqttClient = new MqttClientConnector();
		mqttClient.connect();
		mqttClient.subscribeToTopic(topicName);
		try {
			Thread.sleep(60000);
		} catch (InterruptedException e) {
			// TODO Auto-generated catch block
			e.printStackTrace();
		}
		mqttClient.disconnect();
	}

	
	/**
	 * Main method
	 * 
	 * 	@param args String of arguments
 	*/
	public static void main(String[] args) {
		
		app = new MqttSubClientTestApp();
		String topic = "Temperature Sensor";							// Topic name is set

		try
		{
			app.start(topic);
			String message = MqttClientConnector.getMessag();			//Retrieving Json Data
			logger.info("Received Json Data\n");
			System.out.println("Received Json Message: "+message+"\n");
			DataUtil data = new DataUtil();								
			SensorData sensor = data.JsonToSensorData(message,null);    //Converting Json to SensorData
			logger.info("Printing SensorData:\n");
			System.out.println("After Json to SensorData Conversion:\n"+sensor);
			String json = data.SensorDataToJson(sensor);				//Converting SensorData to Json
			logger.info("Printing Json Data:\n");
			System.out.println("After SensorData to Json Conversion:\n"+json);
		}
		catch (Exception ex)
		{
			ex.printStackTrace();
		}

	}

}
